### PR TITLE
Legacy Consumer: field template improvements

### PR DESCRIPTION
**Pull request by ravinderk**
## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@since` tags included in DocBlocks
-   [x] Changes logged to the `Unreleased` section of `CHANGELOG.md`
-   [ ] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Review by knowler — CHANGES_REQUESTED** [review]
@ravinderk I don’t think we need to change the `select` or `textarea` or `file` templates to wrap the input with the `<label>` element. It makes sense for the `checkbox` since the label appears on the same line and it was visually misaligned (thanks for fixing this). My rule of thumb is to keep the `<label>` element and the form control separate unless they need to be displayed inline in which case I’ll wrap the form control with the `<label>` element. I don’t like needing to put more elements than necessary inside of a `<label>` element, especially since [its content model is phrasing content](https://html.spec.whatwg.org/multipage/forms.html#the-label-element), which does not include `<div>` elements which are commonly used with form controls. In my experience, keeping them separate by default helps set us up to write good HTML.

**Comment by ravinderk** [comment]
@knowler Code updated as per your suggestion.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,12 +7,15 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
 
+###  Changed
+
+- Remove id attribute, add missing attributes and attach label implicitly in custom fields. (#5920)
```
Comment (knowler) [inline CHANGELOG.md:12]: Let’s break these changes apart so we can provide a bit more details:

```suggestion
- Wrap `<input>` element inside of `<label>` element for Legacy Consumer’s checkbox template. (#5920)
- Add missing `required` and `readonly` attributes to Legacy Consumer’s select and textarea templates. (#5920)
- Add screen reader text for required indicator to Legacy Consumer’s label content template. (#5920) 
```

**Review by knowler — CHANGES_REQUESTED** [review]
@ravinderk You’ll need to add the `id` attribute back to the `file`, `textarea`, and `select` templates since they are using the separate `<label>` element again. Other than that and I changelog change I suggested, this looks good.

File: src/Form/LegacyConsumer/templates/label-content.html.php [hunk 2]
```diff
@@ -1,5 +1,8 @@
 <?php /** @var Give\Framework\FieldsAPI\Field $field */ ?>
 <?php echo $field->getLabel(); ?>
 <?php if ( $field->isRequired() ) : ?>
-<span class="give-required-indicator"> *</span>
+	<span class="give-required-indicator">
+		<span aria-hidden="true">*</span>
+		<span class="screen-reader-text"><?php esc_html_e( 'Required', 'give' ); ?></span>
+	</span>
```
Comment (knowler) [inline src/Form/LegacyConsumer/templates/label-content.html.php:7]: Nice work here, @ravinderk!

File: src/Form/LegacyConsumer/templates/textarea.html.php [hunk 3]
```diff
@@ -1,8 +1,7 @@
...
 <textarea
-  name="give_<?php echo $field->getName(); ?>"
-  id="give-<?php echo $field->getName(); ?>"
-	<?php if ( $field->isRequired() ) : ?>
-	required
-  <?php endif; ?>
+	name="give_<?php echo $field->getName(); ?>"
+	id="give-<?php echo $field->getName(); ?>"
+	<?php echo $field->isRequired() ? 'required' : ''; ?>
+	<?php echo $field->isReadOnly() ? 'readonly' : ''; ?>
```
Comment (knowler) [inline src/Form/LegacyConsumer/templates/textarea.html.php:6]: Thanks for catching this.

**Review by knowler — APPROVED** [review]
Great work, @ravinderk! This is ready to merge.

**Pull request merged**